packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -220,36 +220,49 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
         global.accounts.push(...accounts);
       });
 
-      beforeEach(async function () {
-        const basicPlans = await hbarSpendingPlanRepository.findAllActiveBySubscriptionTier(
-          [SubscriptionTier.BASIC],
-          requestDetails,
-        );
-        for (const plan of basicPlans) {
-          await hbarSpendingPlanRepository.delete(plan.id, requestDetails);
-          await ethAddressSpendingPlanRepository.deleteAllByPlanId(plan.id, 'before', requestDetails);
-          await ipSpendingPlanRepository.deleteAllByPlanId(plan.id, 'before', requestDetails);
-        }
+      afterEach(async () => {
+        hbarSpendingPlanRepository.resetAmountSpentOfAllPlans(requestDetails);
+
+        // Note: Since the total HBAR budget is shared across the entire Relay instance by multiple test cases,
+        //       and expense updates occur asynchronously, the wait below ensures that the HBAR amount has sufficient time
+        //       to update properly after each test.
+        await Utils.wait(1500);
       });
 
       describe('@hbarlimiter-batch1 Total HBAR Limit', () => {
-        it('should execute "eth_sendRawTransaction" without triggering HBAR rate limit exceeded', async function () {
-          const parentContract = await deployContract(parentContractJson, accounts[0].wallet);
+        const pollForProperRemainingHbar = async (initialRemainingHbars: number, expectedTxCost: number) => {
+          let updatedRemainingHbars = Number(await metrics.get(testConstants.METRICS.REMAINING_HBAR_LIMIT));
+
+          // Note: expectedTxCost may be retrieved from mirror node which doesn't include the getRecord transaction fee.
+          //       calculating delta = expectedTxCost * tolerance to account for this difference in transaction costs.
+          const delta = expectedTxCost * 0.002;
```

Comment:
Q: why 0.002 and not some other number. It's unclear also could the delta from before be reused?
If not, same thing a targeted config might allow for easy troubleshooting and testing to find the right value without code changes

Reply:
You're right this is indeed unclear! After reviewing logs, I observed that the fee for getting a record is either 83,333 tinybars or 9 tinybars. I then calculated by dividing 83,333 by the `expectedTxCost`—which can vary over time—and arrived at approximately 0.0015, which I rounded up to 0.002 for better flexibility. That’s how 0.002 was determined.

Will update it to use a targeted configuration for improved debugging.

Reply:
Updated. Added TEST_TRANSACTION_RECORD_COST_TOLERANCE config variable which can be reused through out the test.